activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -9,6 +9,9 @@ class NumericalityValidator < EachValidator # :nodoc:
 
       RESERVED_OPTIONS = CHECKS.keys + [:only_integer]
 
+      INTEGER_REGEX = /\A[+-]?\d+\z/
```

Comment:
Regexes in ruby are immutable so I actually don't need to do that. TIL

Reply:
And no, we don't freeze constants just because they are constants. This is defensive programming and have no benefits at all in terms of runtime. If people wants to mutate a constant it is not a freeze that will stop them.